includes/core/classes/class-setup.php

```diff
@@ -679,4 +680,29 @@ public function set_event_archive_labels( array $post_states, WP_Post $post ): a
 		return $post_states;
 	}
 
+	/**
+	 * Display a notification to recommend enabling user registration for GatherPress functionality.
+	 *
+	 * This method checks if user registration is enabled in WordPress settings and displays a
+	 * notification encouraging users to enable registration for optimal GatherPress functionality.
+	 * Users have the option to suppress this notification permanently.
+	 *
+	 * @return void
+	 */
+	public function check_users_can_register() : void {
+		if ( filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) || filter_var( get_option( 'gp_suppress_membership_notification' ), FILTER_VALIDATE_BOOLEAN ) ) {
+			return;
+		}
+
+		if ( null !== filter_input( INPUT_GET, 'action' ) && 'suppress_gp_membership_notification' === filter_input( INPUT_GET, 'action' ) && ! empty( filter_input( INPUT_GET, '_wpnonce' ) ) && wp_verify_nonce( sanitize_text_field( wp_unslash( filter_input( INPUT_GET, '_wpnonce' ) ) ), 'clear-notification' ) ) {
```

Comment:
see comment in paul's PR about lining this up on multiple lines to be more readable.